pkg/apis/pipeline/v1alpha1/pipeline_validation_test.go

```diff
@@ -118,16 +118,25 @@ func TestPipelineSpec_Validate(t *testing.T) {
 	}, {
 		name: "valid parameter variables",
 		p: tb.Pipeline("pipeline", "namespace", tb.PipelineSpec(
-			tb.PipelineParam("baz"),
-			tb.PipelineParam("foo-is-baz"),
+			tb.PipelineParam("baz", v1alpha1.ParamTypeString),
```

Comment:
Similar to Andrea's comment above, we could make this test builder also assume the string type unless a type is passed as an option.

Reply:
That would definitely be better, but do you have any suggestions for how to do that in Go? Andrea's comment I think referenced the fact that the webhook will automatically set the ParamSpec's type field to "string"  if it is nil. 

I could make `PipelineParam(name string, types ...type)` a variadic function so that the type is optional, but then users of the builder can enter as many types as they want and the behavior is unclear (unless I manually check the # of additional parameters and return an error).

Reply:
In that case you could either document that the last one wins, or, yeah, detect the presence of a previous type variadic arg and fail with a descriptive message if you see that. Either way is probably fine, so long as it's documented.

Reply:
So just checked the code : `func PipelineParam(name string, pt v1alpha1.ParamType, ops ...PipelineParamOp)` . Since the variadic part needs to be the final parameter (which is currently a list of operations), I'm not sure how to make ParamType optional.

Reply:
Could I guess make separate builder functions for each type of param..